mscore/musescore.cpp

```diff
@@ -1406,7 +1406,7 @@ void MuseScore::setCurrentScoreView(ScoreView* view)
             synthControl->setScore(cs);
       if (selectionWindow)
             selectionWindow->setScore(cs);
-      if (mixer)
+      if (mixer && cs)
```

Comment:
Doesn't for me.  Did you remember to remove the "&& cs" from the line I commented on?  Otherwise, you get the effect I mentioned - Mixer contains stale pointer.  For me, going back to the original and just replacing the updateAll() call as I suggested (but with corrected syntax - first colon should be question mark) works fine.  If I close the last score, the Mixer shows an empty window.  I suppose an enhancement could be to force the Mixer to to close.